python/dpu_utils/utils/chunkwriter.py

```diff
@@ -81,3 +89,17 @@ def close(self) -> None:
         self.__flush()
         if self.__parallel_writers > 0:
             self.__writer_executors.shutdown(wait=True)
+
+    def __get_max_existing_index(self):
+        """
+        Returns the largest file index within the current output folder.
+        """
+        file_regex = re.compile(f'{self.__file_prefix}([0-9]+){self.__file_suffix}')
+        max_index = 0
+        for file_name in listdir(self.__out_folder.path):

```

Comment:
Since we want to be able to list files even if the `__out_folder` is an `AzurePath`, `listdir` won't do this. Try using `self.__out_folder.iterate_filtered_files_in_dir(f'{self.__file_prefix}*{self.__file_suffix}')`. See [here](https://github.com/microsoft/dpu-utils/blob/master/python/dpu_utils/utils/richpath.py#L195)

I _think_ that this will remove both the need for the regex and will be compatible with all kinds of `RichPath`.